targets.toml

```diff
@@ -0,0 +1,460 @@
+# This file contains all the "targets" cross can be used with by default and is used for generating `src/docker/provided_images.rs` and our images
+# the only required value for a `target` entry is its name in `target.target` and the `os` to use in CI
+#
+# spec is available in `xtask/src/util.rs` on `CiTarget`
+
+[[target]]
+target = "x86_64-apple-darwin"
+os = "macos-12"
+deploy = true
+
+[[target]]
+target = "x86_64-unknown-linux-gnu"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "native qemu-user qemu-system"
+deploy = true
+
+[[target]]
+target = "x86_64-unknown-linux-musl"
+os = "ubuntu-latest"
+std = true
+run = true
+deploy = true
+
+[[target]]
+target = "x86_64-pc-windows-msvc"
+os = "windows-2019"
+deploy = true
+
+[[target]]
+target = "x86_64-unknown-linux-gnu"
+sub = "centos"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "native qemu-user qemu-system"
+
+[[target]]
+target = "aarch64-unknown-linux-gnu"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "qemu-user qemu-system"
+
+[[target]]
+target = "arm-unknown-linux-gnueabi"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+
+[[target]]
+target = "arm-unknown-linux-gnueabihf"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+
+[[target]]
+target = "armv7-unknown-linux-gnueabi"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "qemu-user"
+
+[[target]]
+target = "armv7-unknown-linux-gnueabihf"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "qemu-user qemu-system"
+
+[[target]]
+target = "thumbv7neon-unknown-linux-gnueabihf"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "qemu-user qemu-system"
+
+[[target]]
+target = "i586-unknown-linux-gnu"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+
+[[target]]
+target = "i686-unknown-linux-gnu"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "native qemu-user qemu-system"
+
+[[target]]
+target = "mips-unknown-linux-gnu"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "qemu-user"
+
+[[target]]
+target = "mipsel-unknown-linux-gnu"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "qemu-user qemu-system"
+
+[[target]]
+target = "mips64-unknown-linux-gnuabi64"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+
+[[target]]
+target = "mips64el-unknown-linux-gnuabi64"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "qemu-user qemu-system"
+
+[[target]]
+target = "mips64-unknown-linux-muslabi64"
+os = "ubuntu-latest"
+cpp = true
+dylib = true
+std = true
+run = true
+runners = "qemu-user"
```

Comment:
This is implicit, I know it was part of the original target spec but we can probably remove this.